docs/proposed/psa-driver-interface.md

```diff
@@ -321,6 +321,144 @@ TODO: key input and output for opaque drivers; deterministic key generation for
+2. Lifetime of the `password` is test driver
+- the core calls opaque driver's `"pake_setup"` and subsequent entry points.
+
+### Summary of entry points for PAKE
+
+A PAKE driver has the following entry points:
+* `"pake_setup"` (mandatory): always the first entry point to be called. It is called when all inputs are collected and the computation stage starts. 
+* `"pake_output"` (mandatory): derive cryptographic material for the specified step and output it.
+* `"pake_input"` (mandatory): provides cryptographic material in the format appropriate for the specified step.
+* `"pake_get_implicit_key"` (mandatory): returns implicitly confirmed shared secret from a PAKE.
+* `"pake_abort"` (mandatory): always the last entry point to be called.
+
+For naming purposes, here and in the following subsection, this specification takes the example of a driver with the prefix `"acme"` that implements the PAKE entry point family with a capability that does not use the `"names"` property to declare different type and entry point names. Such a driver must implement the following type and functions, as well as the entry points listed above and described in the following subsections:
+```
+typedef ... acme_pake_operation_t;
+psa_status_t acme_pake_abort( acme_pake_operation_t *operation );
+```
+
+#### PAKE driver inputs
+
+The core conveys the initial inputs for a PAKE operation via an opaque data structure of type `psa_crypto_driver_pake_inputs_t`.
+
+```
+typedef ... psa_crypto_driver_pake_inputs_t; // implementation-specific type
+```
+
+A driver receiving an argument that points to a `psa_crypto_driver_pake_inputs_t` can retrieve its contents by calling one of the get-data functions below.
+
+```
+psa_status_t psa_crypto_pake_get_password(
+    const psa_crypto_driver_pake_inputs_t *inputs,
+    uint8_t **password,
+    size_t *password_len);
+
+psa_status_t psa_crypto_pake_get_role(
+    const psa_crypto_driver_pake_inputs_t *inputs,
+    psa_pake_role_t *role);
+
+psa_status_t psa_crypto_pake_get_cipher_suite(
+    const psa_crypto_driver_pake_inputs_t *inputs,
+    psa_pake_cipher_suite_t *cipher_suite);
+```
+The get-data functions take the following parameters:
+
+The first parameter `inputs` must be a pointer passed by the core to a PAKE driver setup entry point.
+Next parameters are return buffers (must not be null pointers).
+
+These functions can return the following statuses:
+* `PSA_SUCCESS`: value has been successfully obtained
+* `PSA_ERROR_BAD_STATE`: the inputs are not ready
+
+#### PAKE driver setup
+
+```
+psa_status_t acme_psa_pake_setup( acme_pake_operation_t *operation,
+                                  const psa_crypto_driver_pake_inputs_t *inputs );
+```
+
+* `operation` is a zero-initialized operation object.
+* `inputs` is an opaque pointer to the [inputs](#pake-driver-inputs) for the PAKE operation.
+
+The setup driver function should preserve `inputs` for other driver functions.
```

Comment:
I think you mean that it should preserve _the inputs_? Not `inputs` itself (a pointer), nor the immediate contents of `*inputs` (it's an opaque structure).

Reply:
> I think we can do without the copy of the initial inputs by ensuring that the pointer passed to the driver pointing to the initial inputs is valid as long as the core does not call the pake_abort entry point. That way the driver just save the pointer and can access through it the initial inputs.

Actually in this version of the document I meant preserving the pointer to follow @ronald-cron-arm suggestion.
In current implementation (which is inconsistent now with documentation)  `inputs` and driver context are union, so setup function needs to copy all it needs from `inputs` as after writing to driver context the inputs are lost.